libraries/core_mqtt_agent_demo_dependencies.cmake

```diff
@@ -0,0 +1,132 @@
+# coreMQTT Agent module.
+afr_module(NAME core_mqtt_agent INTERNAL )
+
+# Include MQTT library's source and header path variables.
+include("${CMAKE_CURRENT_LIST_DIR}/coreMQTT-Agent/mqttAgentFilePaths.cmake")
+
+# Create a list of all header files in the coreMQTT Agent library.
+# The list of header files will be added to metadata required
+# for the FreeRTOS console.
+set(MQTT_AGENT_HEADER_FILES "")
+foreach(mqtt_agent_public_include_dir ${MQTT_AGENT_INCLUDE_PUBLIC_DIRS})
+    file(GLOB mqtt_agent_public_include_header_files
+              LIST_DIRECTORIES false
+              ${mqtt_agent_public_include_dir}/*.h )
+    list(APPEND MQTT_AGENT_HEADER_FILES ${mqtt_agent_public_include_header_files})
+endforeach()
+
+# Add cmake files of module to metadata.
+afr_module_cmake_files(${AFR_CURRENT_MODULE} 
+    ${CMAKE_CURRENT_LIST_DIR}/coreMQTT-Agent/mqttAgentFilePaths.cmake
+)
+
+afr_module_sources(
+    ${AFR_CURRENT_MODULE}
+    PRIVATE
+        ${MQTT_AGENT_SOURCES}
+        # Header files added to the target so that these are available
+        # in code downloaded from the FreeRTOS console.
+        ${MQTT_AGENT_HEADER_FILES}
+)
+
+afr_module_include_dirs(
+    ${AFR_CURRENT_MODULE}
+    PUBLIC
+        ${MQTT_AGENT_INCLUDE_PUBLIC_DIRS}
+)
+
+# Dependency of module on logging stack and coreMQTT.
+afr_module_dependencies(
+    ${AFR_CURRENT_MODULE}
+    PUBLIC
+        AFR::core_mqtt
+        AFR::logging
+)
+################################################################
+
+# core_mqtt_agent_demo_dependencies module.
+# Metadata module used for the MQTT Agent library in the FreeRTOS console.
+# It represents a collection of module dependencies required
+# by the coreMQTT Agent demos.
+# This module enables the FreeRTOS console experience of enabling
+# library dependencies when selecting the MQTT Agent library, so that the
+# coreMQTT Agent demos can be downloaded.
+afr_module(NAME core_mqtt_agent_demo_dependencies )
+
+afr_set_lib_metadata(ID "core_mqtt_agent_demo_dependencies")
```

Comment:
This metadata configuration and `core_mqtt_agent_dependencies` CMake target are only required if coreMQTT Agent is slated for display on FreeRTOS console.

Reply:
If we want to display the option of downloading the coreMQTT Agent demo for every board that supports a coreMQTT demo, then, we can just utilize the `core_mqtt_agent_demo_dependencies` target for displaying the MQTT library on FreeRTOS console. Thus, we will need to remove the [`core_mqtt_demo_dependencies`](https://github.com/aws/amazon-freertos/blob/202012.00/libraries/core_mqtt_demo_dependencies.cmake#L45-L128) target